be/src/formats/parquet/column_reader.cpp

```diff
@@ -141,6 +146,7 @@ class ListColumnReader : public ColumnReader {
             array_column = down_cast<ArrayColumn*>(nullable_column->mutable_data_column());
         } else {
             DCHECK(dst->is_array());
+            DCHECK(!_field->is_nullable);
```

Comment:
You'd better double check whether this assumption holds.  As we treats all column slots from hive are nullable, and then create columns based on slots property. If the parquet  columns are required, the field parsed from parquet schema will be not nullable. In this case,  the field property may not be same with the columns we created.

Reply:
This DCHECK make sure the parquet field in required, because dst is not a nullable column, it cannot express the semantics of null.